src/components/__tests__/LibrariesPage-test.tsx

```diff
@@ -253,4 +253,25 @@ describe("LibrariesPage", () => {
     expect(wrapper.find(".list .panel").length).to.equal(2);
   });
 
+  it("should filter libraries by attribute", () => {
+    let libraryWithPLS = modifyLibrary(libraries[0], {"pls_id": "12345"}, "basic_info");
+    wrapper.setProps({ libraries: { libraries: [libraryWithPLS, libraries[1]] } });
+    wrapper.setState({ filters: { flipped: false, attributes: { "pls_id": true }}});
+    let filteredLibraries = wrapper.instance().filterByAttribute(wrapper.prop("libraries").libraries);
+    expect(filteredLibraries.length).to.equal(1);
+    expect(filteredLibraries[0]).to.equal(libraryWithPLS);
+  });
+
+  it("should check whether a library has an attribute", () => {
```

Comment:
Since it's a recursive function, can you also give an example with one or two more levels?

Reply:
Given the current structure of `LibraryData`, there's nowhere else for it to go...

Reply:
Yep, that's very true. The function you created is a nice generic function so any test data should be used to test the function itself, and not just in the context of the component. That's what I was thinking. Maybe something like:
```
const testObj = { testlv1: { testlv2: { testlv3: { testlv4: "some value" }}}};
expect(wrapper.instance().hasAttr(testOb, "testlv3")).to.be.true;
expect(wrapper.instance().hasAttr(testOb, "testlv4")).to.be.true;
```

If the `LibraryData` structure changes, you can be sure your function will work. It could also be passed to some util file so any component can use it, but not necessary for now.